File: src.old/classes/Mathc.lua
local Vector3 = require("classes.MathClasses.Vector3")

---@class Mathc
local Mathc = {
    Vector3 = Vector3
}

---@param value number
---@param min number
---@param max number
function Mathc.clamp(value, min, max)
    return math.max(min, math.min(max, value));
end

return Mathc;

File: src/classes/Math/Mathc.ts
export class Mathc {
    static Clamp(value : number, min : number, max : number)
    {
        // transcribed from PHOBOS utilities module
        // uber fast
        if (value < min)
        {
            return min;
        }
        if (value > max)
        {
            return max;
        }
        return value;
    }
    static Sign(value : number)
    {
        if (value < 0)
        {
            return - 1;
        }
        else
        {
            return 1;
        }
    }
}